src/software/ai/hl/stp/tactic/stop_tactic.h

```diff
@@ -1,7 +1,41 @@
 #pragma once
 
-#include "software/ai/hl/stp/action/move_action.h"
+#include "software/ai/hl/stp/action/move_action.h"  // TODO (#1888): remove this dependency
 #include "software/ai/hl/stp/tactic/tactic.h"
+#include "software/ai/intent/stop_intent.h"
+
+struct StopFSM
+{
+    struct Stop
+    { /* state */
+    };
+
+    struct Update
+    {
+        bool coast;
+        TacticUpdate common;
+    };
+
+    auto operator()()
+    {
+        using namespace boost::sml;
+
+        const auto update_stop_intent = [](auto event) {
+            event.common.set_intent(
+                std::make_unique<StopIntent>(event.common.robot.id(), event.coast));
+        };
```

Comment:
Does this function need to be defined in every tactic that wants to use the `stop_intent`?

From what I understand the event is of type `StopFSM::Update`,  so it seems like every tactic is basically going to have the same copy of these `update_.*_intent` actions for their own `.*FSM::Update` struct. 

Is there anyway we can avoid this?

Reply:
I don't think there's anything we can do here